src/main/java/org/opensearch/ad/util/ParseUtils.java

```diff
@@ -456,14 +456,30 @@ public static SearchSourceBuilder addUserBackendRolesFilter(User user, SearchSou
         return searchSourceBuilder;
     }
 
-    public static User getUserContext(Client client) {
+    /**
+     * @deprecated Extensions will not be using the User from the Thread Context. Temporarily use
+     *     {@link #getNullUser()}.
+     */
+    @Deprecated
+    public static UserIdentity getUserContext(Client client) {
         String userStr = client.threadPool().getThreadContext().getTransient(ConfigConstants.OPENSEARCH_SECURITY_USER_INFO_THREAD_CONTEXT);
         logger.debug("Filtering result by " + userStr);
-        return User.parse(userStr);
+        return UserIdentity.parse(userStr);
+    }
+
+    /**
+     * Placeholder method for running AD as an extension without security. User == null when security is disabled or user is super-admin.
+     *
+     * @return null.
+     */
+    public static UserIdentity getNullUser() {
+        // The author of this temporary code thinks it's a really bad idea to have a null object default to super-admin privileges.
+        // But that's how it is so here is the infinity gauntlet.
+        return null;
```

Comment:
For my knowledge, if we are passing null here the role would be super-admin. Will that cause any impact of the functionality of AD plugin? I'm thinking more in a general way where we provide more access to admin as compared to other users

Reply:
I'm not clear what you mean by "impact of the functionality".  It does provide the same level of access as either super-admin or security disabled, which is what we want in the near term.  See the changes in the tests, where a few tests that were supposed to fail for denied access ended up succeeding.

When we have developed security/identity features which include an access token, we will replace these null users with calls associated with the access token, which we will pass to OpenSearch via our REST API/Transport, and the access will be handled there.

Reply:
Those tests answer my question. Thank you!